rewrite-java-21/src/main/java/org/openrewrite/java/isolated/ReloadableJava21TypeMapping.java

```diff
@@ -732,8 +734,8 @@ private void completeClassSymbol(Symbol.ClassSymbol classSymbol) {
         for (Pair<Symbol.MethodSymbol, Attribute> attr : compound.values) {
             Object value = annotationElementValue(attr.snd.getValue());
             JavaType.Method element = requireNonNull(methodDeclarationType(attr.fst, annotType));
-            JavaType.Annotation.ElementValue elementValue = value instanceof Object[] ?
-                    JavaType.Annotation.ArrayElementValue.from(element, ((Object[]) value)) :
+            JavaType.Annotation.ElementValue elementValue = (value instanceof Object[] arrayValue) ?
+                    JavaType.Annotation.ArrayElementValue.from(element, (arrayValue)) :
```

Comment:
We try to keep our java 8, java 11, java 17 and 21 parsers the same as possible. So revert this change please.

Reply:
To add a bit of context: we're unfortunately are having to maintain currently 4, but soon 5 implementations in parallel, each using the particular version compiler internals. A task we do often is fix something in one copy, and then have to replicate that to all other copies, and then it helps us a lot if there's only intentional changes related to compiler internals, for an easier (smaller) diff. As much as we _could_ leverage newer language features in some of these parsers, we've held off for these reasons. Thanks for taking that into account!

Reply:
```suggestion
            JavaType.Annotation.ElementValue elementValue = value instanceof Object[] ?
                    JavaType.Annotation.ArrayElementValue.from(element, ((Object[]) value)) :
```